File: Angular/putrid-grove-angular/src/app/models/item.ts
import ItemPayoff from './itemPayoff';
import User from './user';

export default class Item {
  constructor(
    public id: string,
    public name: string,
    public description: string,
    public createdAt: Date,
    public price: number,
    public iconPath: string,
    public isShared: boolean,
    public isPayedOff: boolean,
    public itemPayOffs: ItemPayoff[],
    public payedBy: User,
    public payedById: string
  ) {}
}

File: DotNet/PutridGroveAPI/PutridGroveAPI/Entities/Item.cs
namespace PutridGroveAPI.Entities;

public class Item
{
    public Guid Id { get; set; }
    public DateTime CreatedAt {get;set; } = DateTime.Now;
    public string Name { get; set; }
    public string Description { get; set; }
    public double Price { get; set; }
    public string IconPath { get; set; }
    public bool IsShared { get; set; }
    public bool IsPayedOff { get; set; }
    public ICollection<ItemPayoff> itemPayOffs {get;set;}
    public User PayedBy {get;set;}
    public Guid PayedById {get;set;}
    
}